frame_metadata/raw/v14.ts

```diff
@@ -0,0 +1,163 @@
+import * as $ from "../../deps/scale.ts"
+import { $ty, $tyId } from "../../scale_info/raw/Ty.ts"
+import { $null, transformTys } from "../../scale_info/transformTys.ts"
+import {
+  blake2_128,
+  blake2_128Concat,
+  blake2_256,
+  Hasher,
+  identity,
+  twox128,
+  twox256,
+  twox64Concat,
+} from "../../util/hashers.ts"
+import { normalizeDocs, normalizeIdent } from "../../util/normalize.ts"
+import { Constant, FrameMetadata, Pallet, StorageEntry } from "../FrameMetadata.ts"
+import { $emptyKey, $partialMultiKey, $partialSingleKey, $storageKey } from "../key_codecs.ts"
+
+const $hasher = $.literalUnion<Hasher>([
+  blake2_128,
+  blake2_256,
+  blake2_128Concat,
+  twox128,
+  twox256,
+  twox64Concat,
+  identity,
+])
+
+const $storageEntry = $.object(
+  $.field("name", $.str),
+  $.field("modifier", $.literalUnion(["Optional", "Default"])),
+  $.taggedUnion("type", [
+    $.variant("Plain", $.field("value", $tyId)),
+    $.variant(
+      "Map",
+      $.field("hashers", $.array($hasher)),
+      $.field("key", $tyId),
+      $.field("value", $tyId),
+    ),
+  ]),
+  $.field("default", $.uint8Array),
+  $.field("docs", $.array($.str)),
+)
+
+const $constant = $.object(
+  $.field("name", $.str),
+  $.field("ty", $tyId),
+  $.field("value", $.uint8Array),
+  $.field("docs", $.array($.str)),
+)
+
+const $pallet = $.object(
+  $.field("name", $.str),
+  $.optionalField(
+    "storage",
+    $.object(
+      $.field("prefix", $.str),
+      $.field("entries", $.array($storageEntry)),
+    ),
+  ),
+  $.optionalField("calls", $tyId),
+  $.optionalField("event", $tyId),
+  $.field("constants", $.array($constant)),
+  $.optionalField("error", $tyId),
+  $.field("i", $.u8),
+)
+
+const $extrinsicDef = $.object(
+  $.field("ty", $tyId),
+  $.field("version", $.u8),
+  $.field(
+    "signedExtensions",
+    $.array($.object(
+      $.field("ident", $.str),
+      $.field("ty", $tyId),
+      $.field("additionalSigned", $tyId),
+    )),
+  ),
+)
+
+// https://docs.substrate.io/build/application-development/#metadata-system
+const magicNumber = 1635018093
+
+export const $metadata = $.object(
+  $.field("magicNumber", $.constant<typeof magicNumber>(magicNumber, $.u32)),
+  $.field("version", $.constant<14>(14, $.u8)),
+  $.field("tys", $.array($ty)),
+  $.field("pallets", $.array($pallet)),
+  $.field("extrinsic", $extrinsicDef),
+)
+
+export function transformMetadata(metadata: $.Native<typeof $metadata>): FrameMetadata {
```

Comment:
Why is this contained within `raw/v14`? Should it be? After all, this gives us the non-raw/processed metadata.

Reply:
Because this transformation logic is specific to `raw/v14` – or in other words, this ensures that the only code that cares about the raw format is contained in the `raw` directory.